### ISSUE TITLE
"jj init --git-repo ." panics with "Git commit '...' already exists with different associated non-Git meta-data"

### ISSUE THREAD

**Issue by yuja** [martinvonz/jj#177]
## Steps to Reproduce the Problem

```
git init testrepo
cd testrepo
touch a
git add .
git ci -m a --date '1970-01-01 00:00:00 +0000'
RUST_BACKTRACE=1 jj init --git-repo .
```
fails with
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Other("Git commit 'c5ddd3afbc7c468a8fe38a823fcc5a358e205858' already exists with different associated non-Git meta-data")', lib/src/store.rs:145:60
stack backtrace:
   0: rust_begin_unwind
             at /usr/src/rustc-1.58.1/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /usr/src/rustc-1.58.1/library/core/src/panicking.rs:107:14
   2: core::result::unwrap_failed
             at /usr/src/rustc-1.58.1/library/core/src/result.rs:1613:5
   3: core::result::Result<T,E>::unwrap
             at /usr/src/rustc-1.58.1/library/core/src/result.rs:1295:23
   4: jujutsu_lib::store::Store::write_commit
             at /home/yuya/work/2022/jj/jj/lib/src/store.rs:145:25
   5: jujutsu_lib::repo::MutableRepo::write_commit
             at /home/yuya/work/2022/jj/jj/lib/src/repo.rs:500:22
   6: jujutsu_lib::commit_builder::CommitBuilder::write_to_repo
             at /home/yuya/work/2022/jj/jj/lib/src/commit_builder.rs:157:22
   7: jujutsu_lib::repo::MutableRepo::check_out
             at /home/yuya/work/2022/jj/jj/lib/src/repo.rs:589:13
   8: jujutsu::commands::cmd_init
             at /home/yuya/work/2022/jj/jj/src/commands.rs:1741:13
   9: jujutsu::commands::dispatch
             at /home/yuya/work/2022/jj/jj/src/commands.rs:4495:37
  10: jj::main
             at /home/yuya/work/2022/jj/jj/src/main.rs:66:26
  11: core::ops::function::FnOnce::call_once
             at /usr/src/rustc-1.58.1/library/core/src/ops/function.rs:227:5
```

It appears that jj tries to create duplicated empty commits by
1. WorkspaceCommandHelper::for_loaded_repo() -> helper.import_git_refs_and_head()
2. tx.mut_repo().check_out()

## Specifications

- Platform: Linux
- Version: 662297acc395 7f4188536bdd

**Comment by martinvonz** [comment]
Hi Yuya, and thanks for your report!

First, if you're just looking for a workaround, you don't need one because the repo is actually left in the state it should be.

This bug is a flavor of #27. As you found, the problem is that `jj init` tries to create two very similar, empty commits. I think the difference between them ("different associated non-Git meta-data") is that the second one has the first one as predecessor. That will no longer happen after I've finished #32 (by making `jj op log --change <change ID>` or something like that replace `jj obslog`).

I think the version of the bug you found here came after I fixed #102, because the checkout added for that is what tries to write the second commit. The fix should be to simply skip that step when we're sharing a working copy with Git, because then the checkout has already been done a little earlier (step 1 in your report).

**Comment by yuja** [comment]
> the version of the bug you found here came after I fixed https://github.com/martinvonz/jj/issues/102, because

Looks like that. I wasn't sure which empty commit would be guilty, but now I can see why test_init_git_colocated() 
has no commit to trigger this bug.

Thanks anyways. I played with jj last weekend, and I love the way of handling working copy (and no git index.)
